plugins/pulp_rpm/plugins/migrations/0043_add_modulemd_modulemd-defaults.py

```diff
@@ -336,14 +336,24 @@ def migrate(*args, **kwargs):
     repos_to_republish = set()
     for file in metadatafiles:
         repository = Repository.objects.get(repo_id=file.repo_id)
-        with gzip.open(file._storage_path, 'r') as fp:
-            working_dir = "/var/cache/pulp"
-            modulemds, defaults = load(fp, working_dir)
-            add_modulemds(repository, modulemds, working_dir)
-            add_defaults(repository, defaults, working_dir)
-            repository_controller.disassociate_units(repository, [file])
-            repository_controller.rebuild_content_unit_counts(repository)
-            repos_to_republish.add(file.repo_id)
+        try:
+            with gzip.open(file._storage_path, 'r') as fp:
+                working_dir = "/var/cache/pulp"
+                modulemds, defaults = load(fp, working_dir)
```

Comment:
The only line which can blow up due to file being uncompressed is this one, right?
During load it reads from file and exception is raised.

Does it make sense to work with files old way without context manager but not duplicate so much code and put into try except only the part we are worried about?

Something like:
```
working_dir = "/var/cache/pulp"

fp = gzip.open(file._storage_path)
try:
    modulemds, defaults = load(fp, working_dir)
except IOError:
    # it turned out to be an uncompressed file
    fp.close()
    fp = open(file._storage_path)
    modulemds, defaults = load(fp, working_dir)
finally:
     fp.close()

add_modulemds(repository, modulemds, working_dir)
repository_controller.disassociate_units(repository, [file])
add_defaults(repository, defaults, working_dir)
repository_controller.rebuild_content_unit_counts(repository)
repository_controller.disassociate_units(repository, [file])
repos_to_republish.add(file.repo_id)
repository_controller.rebuild_content_unit_counts(repository)
repos_to_republish.add(file.repo_id)
```

Reply:
I was just trying to go with the least amount of changes.